plaid/src/loader/mod.rs

```diff
@@ -108,6 +118,15 @@ impl PlaidModule {
             .flatten()
     }
 
+    /// Method to check if the module is currently locked
+    pub fn is_locked(&self) -> bool {
+        if let Some(mutex) = &self.concurrency_unsafe {
+            // Return true if try_lock fails (meaning the mutex is locked), otherwise false
+            return mutex.try_lock().is_err();
```

Comment:
I think there are other errors here than just `WouldBlock`. If the Mutex is poisoned, I don't think we'd ever detect that because this function would always appear to return "Locked" and thus we'd never hit the poison resolution case above.

Reply:
I think the solution here is to log the poison and clear it here.